### PR TITLE
feat: add profiler

### PR DESCRIPTION
**Pull request by ChengYueJia**
* add profiler tool macro

### CODE REVIEW

**Comment by eigmax** [comment]
Hi, Paul, can you plz fix the conflicts and make it ready to be merged?

File: starky/Cargo.toml [hunk 1]
```diff
@@ -38,13 +38,19 @@ pil_analyzer = {git = "https://github.com/powdr-labs/powdr.git", branch = "main"
 number = {git = "https://github.com/powdr-labs/powdr.git", branch = "main", package = "number"}
 ast = {git = "https://github.com/powdr-labs/powdr.git",   branch = "main",package = "ast"}
 
+profiler_macro = {git = "https://github.com/ChengYueJia/profiler-rs", branch = "main"}
```
Comment (eigmax) [inline starky/Cargo.toml:41]: Can we use features=profiler to determine if we import this?

File: starky/src/stark_gen.rs [hunk 2]
```diff
@@ -1120,10 +1120,8 @@ pub mod tests {
             "273030697313060285579891744179749754319274977764",
         )
         .unwrap();
-        end_timer!(start_stark_gen);
-        log::trace!("verify the proof...");
+        log::debug!("verify the proof...");
```
Comment (eigmax) [inline starky/src/stark_gen.rs:1123]: Keep trace level here.

Comment (ChengYueJia) [reply starky/src/stark_gen.rs:1123]: Of course.

File: starky/Cargo.toml (hunk 1 above)
Comment (ChengYueJia) [reply starky/Cargo.toml:41]: Of course.

Comment (eigmax) [reply starky/Cargo.toml:41]: If so, we can set  up optional = true?

File: zkit/Cargo.toml [hunk 3]
```diff
@@ -21,4 +21,9 @@ rand = "0.4"
 
 # error and log
 log = "0.4.0"
-env_logger = "0.10"
\ No newline at end of file
+env_logger = "0.10"
+
+
+[features]
+default = ["profiler"]
```
Comment (eigmax) [inline zkit/Cargo.toml:28]: Can we remove profiler from default feature?

Comment (ChengYueJia) [reply zkit/Cargo.toml:28]: Sure

File: starky/src/fft_worker.rs [hunk 4]
```diff
@@ -107,6 +108,7 @@ fn _fft_block<F: FieldExtension>(
     }
 }
 
+#[time_profiler()]
 pub fn fft_block<F: FieldExtension>(
```
Comment (eigmax) [inline starky/src/fft_worker.rs:112]: Can we remove this profier modifier?  it's mostly run parallelly

File: starky/src/interpreter.rs [hunk 5]
```diff
@@ -195,6 +196,7 @@ impl<T: FieldExtension> fmt::Display for Block<T> {
     }
 }
 
+#[time_profiler()]
```
Comment (eigmax) [inline starky/src/interpreter.rs:199]: This can be removed too.

Comment (ChengYueJia) [reply starky/src/interpreter.rs:199]: Done

**Pull request merged**